lib/IPC/System/Simple.pm

```diff
@@ -419,23 +407,10 @@ sub capturex {
 	# Parent process, we don't care about our pid, but we
 	# do go and read our pipe.
 
-	if ($wantarray) {
-		my @results = <$pipe>;
-		close($pipe);
-		_process_child_error($?,$command,$valid_returns);
-		return @results;
-	}
-
-	# NB: We don't check the return status on close(), since
-	# on failure it sets $?, which we then inspect for more
-	# useful information.
-
```

Comment:
I oppose the deletion of the author's inline comment.  I am now co-maintainer of this distribution and I found this comment helpful.